### PR TITLE
VM/ Folders: Adding search

### PR DESCRIPTION
**Pull request by ngaglani-tenable**
* Added support for search functionality.
* This PR depends upon #69

### CODE REVIEW

**Review by dhvanilshah-tenable — APPROVED** [review]

File: tenable/io/v3/vm/folders/api.py [hunk 1]
```diff
@@ -79,20 +83,95 @@ def edit(self, id: int, name: str) -> None:
...
-            :obj:`list`: List of folder resource records.
-
+            Iterable:
+                The iterable that handles the pagination and potentially
+                async requests for the job.
+            requests.Response:
+                If ``return_json`` was set to ``True``, then a response
+                object is instead returned instead of an iterable.
         Examples:
             >>> for folder in tio.v3.vm.folders.list():
```
Comment (harshshah-tenable) [inline tenable/io/v3/vm/folders/api.py]: tio.v3.vm.folders.search()

**Review by harshshah-tenable — APPROVED** [review]

File: tenable/io/v3/vm/folders/api.py [hunk 2]
```diff
@@ -79,20 +83,95 @@ def edit(self, id: int, name: str) -> None:
...
+                             ('field_name_2', 'desc')]
+                    - [{'property': 'last_observed', 'order': 'desc'}]
+            limit (int):
+                Number of objects to be returned in each request.
+                Default is 200.
+            next (str):
+                The pagination token to use when requesting the next page of
+                results.  This token is presented in the previous response.
+            return_resp (bool):
+                If set to true, will override the default behavior to return
```
Comment (Mangusingh-te) [inline tenable/io/v3/vm/folders/api.py:159]: Update docstring to describe correct behaviour.

File: tenable/io/v3/vm/folders/api.py [hunk 3]
```diff
@@ -79,20 +83,95 @@ def edit(self, id: int, name: str) -> None:
...
+                data in text/csv format on each next call with row headers
+                on each page.
         Returns:
-            :obj:`list`: List of folder resource records.
-
+            Iterable:
+                The iterable that handles the pagination and potentially
+                async requests for the job.
+            requests.Response:
+                If ``return_json`` was set to ``True``, then a response
```
Comment (Mangusingh-te) [inline tenable/io/v3/vm/folders/api.py]: Update docstring to describe correct behavior.

**Review by Mangusingh-te — CHANGES_REQUESTED** [review]

File: tenable/io/v3/vm/folders/api.py (hunk 3 above)
Comment (harshshah-tenable) [reply tenable/io/v3/vm/folders/api.py]: Updated

File: tenable/io/v3/vm/folders/api.py (hunk 2 above)
Comment (harshshah-tenable) [reply tenable/io/v3/vm/folders/api.py:159]: Updated

**Review by alpesh-te — APPROVED** [review]
LGTM!

**Pull request merged**